components/game_table/valorant/game_table_character_custom.lua

```diff
@@ -0,0 +1,174 @@
+---
+-- @Liquipedia
+-- wiki=valorant
+-- page=Module:GameTable/Character/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Arguments = require('Module:Arguments')
+local Array = require('Module:Array')
+local CharacterIcon = require('Module:CharacterIcon')
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+local MathUtil = require('Module:MathUtil')
+local Page = require('Module:Page')
+local Table = require('Module:Table')
+
+local OpponentLibraries = require('Module:OpponentLibraries')
+local Opponent = OpponentLibraries.Opponent
+
+local CharacterGameTable = Lua.import('Module:GameTable/Character')
+
+---@class ValorantCharacterGameTable: CharacterGameTable
+local CustomCharacterGameTable = Class.new(CharacterGameTable, function (self)
+	self.args.showBans = false
+
+	if self.args.tableMode == Opponent.solo then
+		self.isPickedByRequired = true
+		self.args.showOnlyGameStats = true
+		self.statsFromMatches = CharacterGameTable.statsFromMatches
+	end
+end)
+
+---@param game CharacterGameTableGame
+---@return number?
+function CustomCharacterGameTable:getCharacterPick(game)
+	if self.config.mode ~= Opponent.solo then
+		return CharacterGameTable.getCharacterPick(self, game)
+	end
+	local aliases = self.config.aliases
+	local found
+	Table.iter.forEachPair(game.participants, function (participantId, participant)
+		if found then return end
+		if aliases[participant.player] then
+			local pKey = Array.parseCommaSeparatedString(participantId, '_')
+			game.pickedByplayer = tonumber(pKey[2])
+			found = tonumber(pKey[1])
+			return
+		end
+	end)
+
+	return found
+end
+
+---@return integer
+function CustomCharacterGameTable:getNumberOfPicks()
+	return 10
+end
+
+---@param opponentIndex number
+---@param playerIndex number
+---@return string
+function CustomCharacterGameTable:getCharacterKey(opponentIndex, playerIndex)
+	return 't' .. opponentIndex .. 'p' .. playerIndex .. 'agent'
+end
+
+---@return Html
+function CustomCharacterGameTable:headerRow()
+	local makeHeaderCell = function(text, width)
+		return mw.html.create('th'):css('max-width', width):node(text)
+	end
+
+	local config = self.config
+
+	local nodes = Array.append({},
+		makeHeaderCell('Date', '100px'),
+		config.showTier and makeHeaderCell('Tier', '70px') or nil,
+		config.showType and makeHeaderCell('Type', '70px') or nil,
+		config.showIcon and makeHeaderCell(nil, '25px'):addClass('unsortable') or nil,
+		makeHeaderCell('Tournament'),
+		makeHeaderCell('Map'),
+		(config.showResult and config.mode == Opponent.solo) and makeHeaderCell('') or nil,
+		(config.showResult and config.mode ~= Opponent.team) and makeHeaderCell('K') or nil,
+		(config.showResult and config.mode ~= Opponent.team) and makeHeaderCell('D') or nil,
+		(config.showResult and config.mode ~= Opponent.team) and makeHeaderCell('A') or nil,
+		(config.showResult and config.mode ~= Opponent.team) and makeHeaderCell('Ratio') or nil,
```

Comment:
Not sure if this can work, but if it does this might be neat. Similar with the ones only depending on showResult
```suggestion
		unpack(config.showResult and config.mode ~= Opponent.team) and 
		  {
		    makeHeaderCell('K'),
		    makeHeaderCell('D'),
		    makeHeaderCell('A'),
		    makeHeaderCell('Ratio')
		  } or {}
```

Reply:
Tried https://liquipedia.net/valorant/index.php?title=Module%3AGameTable%2FCharacter%2FCustom&type=revision&diff=996623&oldid=996621, but only th `K` would appear.

Reply:
Interesting, did some testing and it seems to break as soon as additional arguments come after the unpack.
Sadly Array.extend which could handle tables in the input also does not work due to the return value of makeHeaderCell being a node (and therefore a table)

Reply:
Yap, lets keep it simple, I guess ...